src/lib/network/edit_nic_name.rb

```diff
@@ -19,33 +19,17 @@ class EditNicName
     # @return [String] current udev match criteria
     attr_reader :old_key
 
-    # udev rule attribute for MAC address
-    MAC_UDEV_ATTR   = "ATTR{address}".freeze
-
-    # udev rule attribute for BUS id
-    BUSID_UDEV_ATTR = "KERNELS".freeze
-
-    def initialize
+    # Constructor
+    #
+    # @param settings [Y2Network::InterfaceConfigBuilder] Interface configuration
+    def initialize(settings)
       textdomain "network"
-
-      Yast.include self, "network/routines.rb"
-
-      current_item = LanItems.getCurrentItem
-      current_rule = LanItems.current_udev_rule
-
-      @old_name = LanItems.current_udev_name
-      unless current_rule.empty?
-        @old_key = :mac unless LanItems.GetItemUdev(MAC_UDEV_ATTR).empty?
-        @old_key = :bus_id unless LanItems.GetItemUdev(BUSID_UDEV_ATTR).empty?
-      end
-
-      if current_item["hwinfo"]
-        @mac = current_item["hwinfo"]["permanent_mac"]
-        @bus_id = current_item["hwinfo"]["busid"]
-      else
-        @mac = ""
-        @bus_id = ""
-      end
+      @settings = settings
+      interface = settings.interface
+      @old_name = interface.name
+      @old_key = interface.renaming_mechanism
+      @mac = interface.hardware.mac
```

Comment:
what happens when hardware is nil? ( sadly we now call it even for fake devices )

Reply:
I will fix this in a separate PR. Actually, the widget must be updated to display a textfield where the user can specify the value